web-app/CKS.Web.Test/IntegrationTests/SearchTests.cs

```diff
@@ -11,19 +18,67 @@ public class SearchTests : TestBase
 		public SearchTests(ITestOutputHelper output) : base(output)
 		{ }
 
-		[Obsolete("Please replace this test with a real test when the search endpoint is setup")]
+
+		private HttpClient SetupClientWithFakeSearchProvider(string fakeSearchResultsFilePath = null)
+		{
+			var searchResults = new SearchResults();
+			if(fakeSearchResultsFilePath != null)
+				searchResults = JsonConvert.DeserializeObject<SearchResults>(File.ReadAllText(fakeSearchResultsFilePath));
+
+			var mockSearchProvider = new Mock<ISearchProvider>();
+			mockSearchProvider.Setup(x => x.Search(It.IsAny<ISearchUrl>())).Returns(searchResults);
+			var client = _factory
+				.WithImplementation<ISearchProvider>(mockSearchProvider.Object)
+				.CreateClient();
+
+			return client;
+		}
+
 		[Fact]
 		public async void ReturnsJson()
 		{
 			// Arrange
-			var client = _factory
-			   .CreateClient();
+			var client = SetupClientWithFakeSearchProvider();
 
 			// Act
 			var response = await client.GetAsync("/api/search");
 
+			// Assert
 			response.StatusCode.ShouldBe(System.Net.HttpStatusCode.OK);
 			response.Content.Headers.ContentType.MediaType.ShouldBe("application/json");
 		}
+
+
+		[Fact]
+		public async void SearchForCancerReturnsCorrectJson()
+		{
+			// Arrange
+			var client = SetupClientWithFakeSearchProvider(@"./IntegrationTests/Fakes/FakeSearchResult-Cancer.json");
+
+			//Act
+			var response = await client.GetAsync("/api/search?q=cancer");
+			var responseBody = await response.Content.ReadAsStringAsync();
+			responseBody = Formatter.FormatJson(responseBody);
+
+			//Assert
+			response.StatusCodeIs(HttpStatusCode.OK);
+			responseBody.ShouldMatchApproved();
```

Comment:
There definitely value in these tests alongside the unit tests. The unit tests don't test a couple of things that the integration tests do (or can):

1. The routing - we want to assert that results are returned from the /api/search endpoint
2. Serialization - the unit tests are asserting on the object returned from the action method directly and not the actual response
3. Response caching (either via a middleware or action method attribute)

However, these tests don't need to be approvals tests: at the moment they're asserting that the json is as expected, but it makes a lot more sense to assert that the returned body of the response equals the original json file we loaded into the search provider. As in, we don't actually care here what the values are, as long as the response matches it - the JSON could be `{"dylan": true}` for all we care.